File: app/src/main/java/com/fos/fosmvp/base/BasePresenter.java
package com.fos.fosmvp.base;

import android.content.Context;

import com.fos.fosmvp.baserx.RxManager;


/**
 * 基类presenter
 */
public abstract class BasePresenter<T,E>{
    public Context mContext;
    public T mView;
    public E mModel;
    public RxManager mRxManage = new RxManager();

    public void setViewModel(T v, E m) {
        this.mView = v;
        this.mModel = m;
        this.onStart();
    }
    public void onStart(){

    }
    public void onDestroy() {
        mRxManage.clear();
    }
}

File: app/src/main/java/com/fos/fosmvp/common/base/BasePresenter.kt
package com.fos.fosmvp.common.base

import android.content.Context

import com.fos.fosmvp.common.baserx.RxManager


/**
 * 基类presenter
 */
abstract class BasePresenter<T, E> {
    var mContext: Context? = null
    var mView: T? = null
    var mModel: E? = null
    var mRxManage = RxManager()

    fun setViewModel(v: T, m: E) {
        this.mView = v
        this.mModel = m
        this.onStart()
    }

    fun onStart() {

    }

    fun onDestroy() {
        mRxManage.clear()
    }
}
